hadoop-ozone/dist/src/main/compose/upgrade/testlib.sh

```diff
@@ -106,7 +106,7 @@ run_test() {
 
   local test_dir="$_upgrade_dir/upgrades/$upgrade_type"
   local callback_dir="$test_dir"/callbacks
-  local execution_dir="$test_dir"/execution/"${OZONE_UPGRADE_FROM}-${OZONE_UPGRADE_TO}"
+  local execution_dir="$test_dir"/execution/"$compose_cluster"/"${OZONE_UPGRADE_FROM}-${OZONE_UPGRADE_TO}"
```

Comment:
How about:
```suggestion
  local execution_dir="$test_dir"/execution/"$compose_cluster-$upgrade_type-${OZONE_UPGRADE_FROM}-${OZONE_UPGRADE_TO}"
```

This would fit better if we have other upgrade type in future.

Reply:
Hi @peterxcli ,

Thanks for the suggestion. Current change on line 109 **execution_dir** is based on line 107 **test_dir**, which already include $upgrade_type as folder. Therefore I think might not need to concatenate $upgrade_type with $compose_cluster as new folder name.

Reply:
Both `$upgrade_type` and `$compose_cluster` are included in the full path, so files are not overwritten when executing the test.  However, only the name of the leaf directory is used when moving results/outputs of each execution to the final directory.  Therefore, the artifact still does not have test results for both executions.  You can verify this by comparing the contents of the [artifact for this change](https://github.com/Jimmyweng006/ozone/actions/runs/14265344587/artifacts/2883270133) vs. [one on `master` branch](https://github.com/apache/ozone/actions/runs/14265561966/artifacts/2882997277): they have exactly the same files.

Reply:
Hi @adoroszlai ,

I see that both artifacts' results end up in the same directory because the copy_results() function only uses the basename of the execution directory. I wasn’t aware of this behavior before.

Thanks for the review and for pointing this out.